templates/posts/details.rs.html

```diff
@@ -0,0 +1,146 @@
+    }
+    <article>
+        @Html(&article.content)
+    </article>
+
+    <div class="article-meta">
+        <p>@i18n!(ctx.1, "This article is under the {0} license."; &article.license)</p>
+        <ul class="tags">
+            @for tag in tags {
+		        @if !tag.is_hashtag {
+                    <li><a href="/tag/@tag.tag">@tag.tag</a></li>
+                }
+            }
+        </ul>
+        <div class="flex">
+            @avatar(ctx.0, &author, Size::Medium, true, ctx.1)
+            <div class="grow">
+                <h2><a href="/@@/@author.get_fqn(ctx.0)">@author.name(ctx.0)</a></h2>
+                <p>@Html(&author.summary)</h2>
+            </div>
+            <a href="/@@/@author.get_fqn(ctx.0)/follow" class="button">
+                @if is_following {
+                    @i18n!(ctx.1, "Unfollow")
+                } else {
+                    @i18n!(ctx.1, "Follow")
+                }
+            </a>
+        </div>
+
+        @if ctx.2.is_some() {
+            <div class="actions">
+                <form class="likes" action="@article.url(ctx.0)/like" method="POST">
+                    <p aria-label="@i18n!(ctx.1, "One like", "{0} likes", &n_likes)" title="@i18n!(ctx.1, "One like", "{0} likes", n_likes)">
+                        @n_likes
+                    </p>
+
+                    @if has_liked {
+                        <button type="submit" class="action liked">@icon!("heart") @i18n!(ctx.1, "I don't like this anymore")</button>
+                    } else {
+                        <button type="submit" class="action">@icon!("heart") @i18n!(ctx.1, "Add yours")</button>
+                    }
+                </form>
+                <form class="reshares" action="@article.url(ctx.0)/reshare" method="POST">
+                    <p aria-label="@i18n!(ctx.1, "One boost", "{0} boost", &n_reshares)" title="@i18n!(ctx.1, "One boost", "{0} boosts", n_reshares)">
+                        @n_reshares
+                    </p>
+
+                    @if has_reshared {
+                        <button type="submit" class="action reshared">@icon!("repeat") @i18n!(ctx.1, "I don't want to boost this anymore")</button>
+                    } else {
+                        <button type="submit" class="action">@icon!("repeat") @i18n!(ctx.1, "Boost")</button>
+                    }
+                </form>
+            </div>
+        } else {
+            <p class="center">@i18n!(ctx.1, "Login or use your Fediverse account to interact with this article")</p>
+            <div class="actions">
+                <div class="likes">
+                    <p aria-label="@i18n!(ctx.1, "One like", "{0} likes", &n_likes)" title="@i18n!(ctx.1, "One like", "{0} likes", n_likes)">
+                        @n_likes
+                    </p>
+                    <a href="/login?m=Login%20to%20like" class="action">@icon!("heart") @i18n!(ctx.1, "Add yours")</a>
```

Comment:
Message won't get translated

Reply:
actually as there are route for it, and has_liked and has_reshared are both set to false by default, the whole `@if ctx.2.is_some() {} else {}` could be turned into a single branch. That way there is less code and it work better. The only issue is there would be no csrf token (no session cookie so rocket_csrf don't issue a token), so the user would be faced with an invalid csrf page. I should probably make rocket_csrf accept request with no cookie, not only response. Anyway it make no sens to do a csrf attack on a client having no session. And this would also remove most exceptions we set in main